cishouseholds/pipeline/run.py

```diff
@@ -84,8 +84,48 @@ def run_pipeline_stages(pipeline_stage_list: list, run_id: int, retry_count: int
         print(
             f"    - completed in: {stage_run_time//60:.0f} minute(s) and {stage_run_time%60:.1f} second(s) in {attempt} attempt(s)"  # noqa:E501
         )  # functional
+
+        splunk_logging(
+            run_id=run_id,
+            stage_name=stage_name,
+            event_time=attempt_start,
+            exception=Exception,
+            success_run=stage_success,
+        )
     return pipeline_error_count
 
 
+def splunk_logging(
+    run_id,
+    stage_name,
+    event_time,
+    exception,
+    success_run,
+    output_directory="",
+):
+    """
+    This function executes at every stage writing in a log file in append mode the run_id,
+    stage_name, event_start_time and status.
+
+    Parameters
+    ----------
+    run_id
+    stage_name
+    event_time
+    exception
+    success_run
+    output_directory=""
+    """
+    with open(f"{output_directory}/status.log", "a") as f:
+        if success_run is True:
```

Comment:
Please adapt the function to be generic, rather than just for logging status. It should take `**kwargs` as it's last parameter and should insert the key value pairs into the output string

Things like status and error message can then be determined outside of the function and passed in

The only two permanent values should be the `event_time` and `run_id`. Please rename to "event_time" below. You could use `get_run_id()` to avoid needing to pass the id in here too